### PR TITLE
Update main branch to 2.9

### PR DESCRIPTION
**Pull request by alecgrieser**
This merges in the `fdb-record-layer-develop-2.9` branch into the main branch, which updates the current minor version of the project to 2.9.

The only change in this PR that is not in any of the 2.9 commits are to fix some merge conflicts in the release notes.

### CODE REVIEW

File: docs/ReleaseNotes.md [hunk 1]
```diff
@@ -100,6 +115,8 @@ The `FDBDatabase::getReadVersion()` method has been replaced with the `FDBRecord
 
 * **Bug fix** Opening a store with `ERROR_IF_NO_INFO_AND_HAS_RECORDS_OR_INDEXES` now allows opening a record store if the data is in the `INDEX_BUILD_SPACE` [(Issue #941)](https://github.com/FoundationDB/fdb-record-layer/issues/941)
 
+=======
+>>>>>>> upstream/fdb-record-layer-develop-2.9
```
Comment (MMcM) [inline docs/ReleaseNotes.md:119]: Uhm, that looks suspicious.

Comment (alecgrieser) [reply docs/ReleaseNotes.md:119]: Oh, whoops. Good point

**Comment by alecgrieser** [comment]
There's also some merge skew in the `RankIndexMaintainer`. Working on resolving that.

**Comment by alecgrieser** [comment]
Hm, well, the sonar build failed with:

```
 > Task :sonarqube FAILED
23:33:02 SonarQube server [https://sonarcloud.io] can not be reached
```

test this please

**Comment by alecgrieser** [comment]
Okay, some prbs failed with some errors that are essentially saying there were too many builds running at once. So, let's try this one more time.

test this please

**Comment by alecgrieser** [comment]
More builds failures from the build environment.

test this please

**Pull request merged**